frame/support/procedural/src/pallet/expand/call.rs

```diff
@@ -0,0 +1,198 @@
+		.collect::<Vec<_>>();
+
+	let args_type = def.call.methods.iter()
+		.map(|method| method.args.iter().map(|(_, _, type_)| type_.clone()).collect::<Vec<_>>())
+		.collect::<Vec<_>>();
+
+	let args_compact_attr = def.call.methods.iter().map(|method| {
+		method.args.iter()
+			.map(|(is_compact, _, _)| {
+				if *is_compact {
+					quote::quote!( #[codec(compact)] )
+				} else {
+					quote::quote!()
+				}
+			})
+			.collect::<Vec<_>>()
+	});
+
+	let args_metadata_type = def.call.methods.iter().map(|method| {
+		method.args.iter()
+			.map(|(is_compact, _, type_)| {
+				if *is_compact {
+					format!("Compact<{:?}>", type_)
+				} else {
+					format!("{:?}", type_)
+				}
+			})
+			.collect::<Vec<_>>()
+	});
+
+	quote::quote_spanned!(call_item_span =>
+		#[cfg_attr(feature = "std", derive(#scrate::DebugNoBound))]
+		#[cfg_attr(not(feature = "std"), derive(#scrate::DebugStripped))]
+		#[derive(
+			#scrate::CloneNoBound,
+			#scrate::EqNoBound,
+			#scrate::PartialEqNoBound,
+			#scrate::codec::Encode,
+			#scrate::codec::Decode,
+		)]
+		#[allow(non_camel_case_types)]
+		pub enum #call_ident<#type_decl_gen> {
+			#[doc(hidden)]
+			#[codec(skip)]
+			__Ignore(
+				#scrate::sp_std::marker::PhantomData<(#type_use_gen,)>,
+				#scrate::Never,
+			),
+			#( #fn_( #( #args_compact_attr #args_type )* ), )*
+		}
+
+		impl<#type_impl_gen> #scrate::dispatch::GetDispatchInfo for #call_ident<#type_use_gen>
+			#where_clause
+		{
+			fn get_dispatch_info(&self) -> #scrate::dispatch::DispatchInfo {
+				match *self {
+					#(
+						Self::#fn_ ( #( ref #args_name, )* ) => {
+							let base_weight = #fn_weight;
+
+							let weight = <
+								dyn #scrate::dispatch::WeighData<( #( & #args_type, )* )>
+							>::weigh_data(&base_weight, ( #( #args_name, )* ));
+
+							let class = <
+								dyn #scrate::dispatch::ClassifyDispatch<( #( & #args_type, )* )>
+							>::classify_dispatch(&base_weight, ( #( #args_name, )* ));
+
+							let pays_fee = <
+								dyn #scrate::dispatch::PaysFee<( #( & #args_type, )* )>
+							>::pays_fee(&base_weight, ( #( #args_name, )* ));
+
+							#scrate::dispatch::DispatchInfo {
+								weight,
+								class,
+								pays_fee,
+							}
+						},
+					)*
+					Self::__Ignore(_, _) => unreachable!("__Ignore cannot be used"),
+				}
+			}
+		}
+
+		impl<#type_impl_gen> #scrate::dispatch::GetCallName for #call_ident<#type_use_gen>
+			#where_clause
+		{
+			fn get_call_name(&self) -> &'static str {
+				match *self {
+					#( Self::#fn_(..) => stringify!(#fn_), )*
+					Self::__Ignore(_, _) => unreachable!("__PhantomItem cannot be used."),
+				}
+			}
+
+			fn get_call_names() -> &'static [&'static str] {
+				&[ #( stringify!(#fn_), )* ]
+			}
+		}
+
+		impl<#type_impl_gen> #scrate::traits::UnfilteredDispatchable for #call_ident<#type_use_gen>
+			#where_clause
+		{
+			type Origin = frame_system::pallet_prelude::OriginFor<T>;
+			fn dispatch_bypass_filter(
+				self,
+				origin: Self::Origin
```

Comment:
An emtpy call e.g. `#[pallet::call] impl<T: Trait> Call for Module<T> {}` will result in `warning: unused variable: origin`